shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/plants/presentation/home/HomeScreenComponent.kt

```diff
@@ -20,7 +21,14 @@ class HomeScreenComponent(
     fun onEvent(event: HomeScreenEvent) {
         when (event) {
             is HomeScreenEvent.OnItemClicked -> onNavigateToDetail(event.item)
-            is HomeScreenEvent.OnFavoriteButtonClicked -> Unit //TODO()
+            is HomeScreenEvent.OnFavoriteButtonClicked -> {
+                _state.updateAndGet { it.copy(plants = it.plants.map { plant ->
+                    plant.takeUnless { plant.id == event.item.id }?: plant.copy(isFavorite = !plant.isFavorite) })
+                }
```

Comment:
You won't need these lines. I think it is better if the UI or your HomeScreenState reacts to the stored favourite state, so If you do something similar here as you did on favouriteScreen it would be the best. combine plants and favourites and when the user toggles the fav state it will emit a new list of favourite ids.

Reply:
Fixed